lib/Target/LLVMIR/LLVMDIScope.cpp

```diff
@@ -90,7 +90,10 @@ struct LLVMDIScopePass : public impl::LLVMDIScopeBase<LLVMDIScopePass> {
         compileUnitAttr = LLVM::DICompileUnitAttr::get(
             distinctId, llvm::dwarf::DW_LANG_C, fileAttr,
             StringAttr::get(context, "triton"),
-            /*isOptimized=*/true, LLVM::DIEmissionKind::LineTablesOnly);
+            /*isOptimized=*/true,
+            LLVM::DIEmissionKind::Full); // DIEmissionKind::Full is required by
```

Comment:
I think we will need a separate Triton compilation mode for full emission. Right now, we only emit line info but full debug emission may have some other consequences. IIRC, there are two modes of "full" debuginfo emission in PTX, one that affects perf and one that does not.

Reply:
@Mogball Could this be gated by the LLVM_EXTRACT_DI_LOCAL_VARIABLES env var?